istioctl/cmd/istioctl/kubeinject.go

```diff
@@ -77,6 +78,29 @@ func getMeshConfigFromConfigMap(kubeconfig string) (*meshconfig.MeshConfig, erro
 	return cfg, err
 }
 
+// grabs the raw values from the ConfigMap. These are encoded as JSON.
+func getValuesFromConfigMap(kubeconfig string) (string, error) {
+	client, err := createInterface(kubeconfig)
+	if err != nil {
+		return "", err
+	}
+
+	config, err := client.CoreV1().ConfigMaps(istioNamespace).Get(injectConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("could not find valid configmap %q from namespace  %q: %v - "+
```

Comment:
IMO it would be cleaner and simpler to just use local files, and not require k8s access to the cluster. 

We can't break compat - the old code getting the template from cluster probably needs to stay - but we shouldn't add more.

kubeinject should work in a CI/CD without access to cluster.

Reply:
`--valuesFile` flag exists, this works like the injection config - optional flag to specify local file, and fallback to configmap.

Is that what you mean or something more?